packages/pages/src/vite-plugin/build/buildStart/rendering/wrapper.ts

```diff
@@ -56,18 +57,20 @@ const getCssTags = (
   filepath: string,
   manifest: bundlerManifest,
   seen: Set<string>
-) => {
-  const entry = Object.entries(manifest).find(([file]) => file === filepath);
+): Set<string> => {
+  const entry = structuredClone(
+    Object.entries(manifest).find(([file]) => file === filepath)
+  );
   if (!entry) {
-    return [];
+    return new Set();
   }
   const [file, info] = entry;
 
   seen.add(file);
-  const cssFiles = info.css || [];
+  const cssFiles = new Set(info.css);
   (info.imports || [])
     .flatMap((f) => getCssTags(f, manifest, seen))
-    .forEach((f) => cssFiles.push(f));
+    .map((f) => f.forEach((file) => cssFiles.add(file)));
```

Comment:
this nested map seems unnecessary since the above line is flat-mapping? If it is necessary then the outer call should also be .forEach, not map.

Reply:
flatmap doesn't work on sets. changed it to be cleaner